lib/gnat.ex

```diff
@@ -367,6 +379,9 @@ defmodule Gnat do
   @impl GenServer
   def init(connection_settings) do
     connection_settings = Map.merge(@default_connection_settings, connection_settings)
+    if :ets.whereis(:endpoint_stats) == :undefined do
```

Comment:
Do you think we should setup the ETS table in the `ConsumerSupervisor` when we are doing the `initialize_as_service` call? That way we don't create the ETS table for people who aren't using services?

Maybe setting up an empty ETS table doesn't really matter? I don't think it's much overhead

Reply:
Yeah we could probably move it. As for overhead, I think an empty ETS table is just a `ref` and shouldn't take much.

Reply:
Pushed the refactor